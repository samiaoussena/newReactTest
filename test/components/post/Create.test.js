jest.mock('mapbox');
import React from 'react';
import renderer from 'react-test-renderer';

import Createjob from '../../../src/components/job/Create';

describe('Createjob', () => {
    test('snapshot', () => {
        const props = { onSubmit: jest.fn() };
        const component = renderer.create(<Createjob {...props} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('handlejobChange', () => {
        const props = { onSubmit: jest.fn() };
        const mockEvent = { target: { value: 'value' } };
        Createjob.prototype.setState = jest.fn();
        const component = new Createjob(props);
        component.handlejobChange(mockEvent);
        expect(component.setState).toHaveBeenCalled();
    });
    test('handleRemoveLocation', () => {
        const props = { onSubmit: jest.fn() };
        Createjob.prototype.setState = jest.fn();
        const component = new Createjob(props);
        component.handleRemoveLocation();
        expect(component.setState).toHaveBeenCalled();
    });
    test('handleSubmit', () => {
        const props = { onSubmit: jest.fn() };
        const mockEvent = {
            target: { value: 'value' },
            preventDefault: jest.fn()
        };
        Createjob.prototype.setState = jest.fn();
        const component = new Createjob(props);
        component.state = {
            valid: true,
            content: 'content',
            location: 'place',
            locationSelected: true
        };
        component.handleSubmit(mockEvent);
        expect(component.setState).toHaveBeenCalled();
        expect(props.onSubmit).toHaveBeenCalledWith({
            content: 'content',
            location: 'place'
        });
    });
    test('onLocationUpdate', () => {
        const props = { onSubmit: jest.fn() };
        Createjob.prototype.setState = jest.fn();
        const component = new Createjob(props);
        component.onLocationUpdate({});
        expect(component.setState).toHaveBeenCalled();
    });
    test('handleToggleLocation', () => {
        const props = { onSubmit: jest.fn() };
        const mockEvent = {
            preventDefault: jest.fn()
        };
        Createjob.prototype.setState = jest.fn();
        const component = new Createjob(props);
        component.handleToggleLocation(mockEvent);
        expect(component.setState).toHaveBeenCalled();
        expect(mockEvent.preventDefault).toHaveBeenCalled();
    });
    test('onLocationSelect', () => {
        const props = { onSubmit: jest.fn() };
        Createjob.prototype.setState = jest.fn();
        const component = new Createjob(props);
        component.onLocationSelect('location');
        expect(component.setState).toHaveBeenCalled();
    });
    test('renderLocationControls', () => {
        const props = { onSubmit: jest.fn() };
        const component = renderer.create(<Createjob {...props} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
