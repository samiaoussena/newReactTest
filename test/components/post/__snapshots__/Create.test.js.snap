// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Createjob renderLocationControls 1`] = `
<div
  className="create-job"
>
  <textarea
    maxlength="280"
    onChange={[Function]}
    placeholder="What's on your mind?"
    value=""
  />
  <div
    className="controls"
  >
    <button
      onClick={[Function]}
    >
      job
    </button>
    <button
      className="open"
      onClick={[Function]}
    >
      Add location
       
      <i
        className="fa fa-map-o"
      />
    </button>
  </div>
  <div
    className="location-picker"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="location-typeahead"
    >
      <i
        className="fa fa-location-arrow"
        onClick={[Function]}
      />
      <input
        onChange={[Function]}
        placeholder="Enter a location..."
        type="text"
        value=""
      />
      <button
        className="open"
        disabled={true}
        onClick={[Function]}
      >
        Select
      </button>
    </div>
    <div
      className="displayMap"
    >
      <div
        className="map"
      />
    </div>
  </div>
</div>
`;

exports[`Createjob snapshot 1`] = `
<div
  className="create-job"
>
  <textarea
    maxlength="280"
    onChange={[Function]}
    placeholder="What's on your mind?"
    value=""
  />
  <div
    className="controls"
  >
    <button
      onClick={[Function]}
    >
      job
    </button>
    <button
      className="open"
      onClick={[Function]}
    >
      Add location
       
      <i
        className="fa fa-map-o"
      />
    </button>
  </div>
  <div
    className="location-picker"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="location-typeahead"
    >
      <i
        className="fa fa-location-arrow"
        onClick={[Function]}
      />
      <input
        onChange={[Function]}
        placeholder="Enter a location..."
        type="text"
        value=""
      />
      <button
        className="open"
        disabled={true}
        onClick={[Function]}
      >
        Select
      </button>
    </div>
    <div
      className="displayMap"
    >
      <div
        className="map"
      />
    </div>
  </div>
</div>
`;
